crypto/s390xcap.c

```diff
@@ -32,7 +56,7 @@ void OPENSSL_cpuid_setup(void)
     if (OPENSSL_s390xcap_P[0])
         return;
 
-    OPENSSL_s390xcap_P[0] = 1UL << (8 * sizeof(unsigned long) - 1);
+    OPENSSL_s390xcap_P[0] = 1ULL << (8 * sizeof(unsigned long long) - 1);
```

Comment:
On related *side* note one can wonder if it makes sense to introduce support for OPENSSL_s390xcap environment variable. Though with the number of options, controlling vector by numbers would be too tricky. How about specifying processors instead? I mean it would recognize values like "z990", "z10", "z196", etc. using them to mask capabilities of current processor...

["Side note" means that no immediate action is expected, it's rather an invitation to discussion of an idea that might be implemented later.]

Reply:
I already implemented an OPENSSL_s390xcap env variable (see old PR). If you would like to review these patches next, i can make this my next PR.

The implementation allows controlling single bits in the vector and i agree its tricky. However, since this is more of a "developer option", id prefer to have this fine-grained control.
Heres an example how i use it to test and measure performance of all available code paths:
https://github.com/p-steuer/ossl-test/blob/master/s390x/config.pm

Reply:
> I already implemented an OPENSSL_s390xcap env variable

Ah! I recall noting it, but didn't look into implementation details.

> ... more of a "developer option"

Yes, but it's also a "support option" and maybe more so, because normally there are more users than developers :-) Just consider a bug report filed to Debian. Which option would user and Debian people appreciate more? Besides, it makes lesser sense to run all tests with *all* combination of flags. I mean if you flip SHA bits, it doesn't affect AES test outcome. So I'd still argue in favour of "package" options denoting processor generations. This doesn't actually preclude possibility of combining the two. I mean it can recognize both ways.

Reply:
> config.pm

Though I'm still kind of opposed to this. How about notation in `kimd:~0x????;km:~0x????` style? I mean instead of implicit indices calculated by amount of colons, you'd *name* the fields.

Well, this is something for new PR, so from now on let's concentrate on things other than environment variable...

Reply:
id also like to have a bit-level control from a "support option" point of view: imagine a single function code not behaving as expected , forcing to step back one machine generation regarding capabilities.

however , im of course not opposed to having a more user-friendly shortcut notation in addition, where ie "z13" would be expanded to the appropriate combined bit mask vector.

Yes, this should be a new PRs subject, we dont have to agree on the env var thing here and now .